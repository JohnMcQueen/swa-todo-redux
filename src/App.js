import React from 'react';

function App() {
  return (
    <div className="App">
      Hello SWA!
    </div>
  );
}

export default App;
