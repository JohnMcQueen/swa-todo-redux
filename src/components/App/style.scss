.app {
  &--snackbar {
      &_successful {
          // please forgive the !important. I didn't want to do the standard
          // material ui override as it's outside the scope of our sessions
          background-color: green !important;
      }

      &_failure {
          background-color: red !important;
      }
  }
}
