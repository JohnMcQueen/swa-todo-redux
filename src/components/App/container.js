import App from './component';
import {connect} from 'react-redux';
import {getUser, selectUser} from 'modules/Users'

const mapStateToProps = (state) => ({
    user: getUser(state)
});

const mapDispatchToProps = (dispatch) => ({
    selectUser: (user) => dispatch(selectUser(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);