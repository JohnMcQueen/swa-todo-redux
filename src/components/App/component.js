import Container from '@material-ui/core/Container';
import TodoList from '../TodoList';
import Typography from '@material-ui/core/Typography';
import React from 'react';

import './style.scss';

export default () => {
  return (
    <Container className="app"> 
      <Typography variant="h2">
        Todo
      </Typography>
      <TodoList></TodoList>
    </Container>
  ); 
};