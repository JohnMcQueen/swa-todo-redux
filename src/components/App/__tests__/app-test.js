import App from 'components/App/component';
import React from 'react';
import {shallow} from 'enzyme';

test('<App />', () => {
  const appWrapper = shallow(<App />);

  expect(appWrapper).toMatchSnapshot();
})