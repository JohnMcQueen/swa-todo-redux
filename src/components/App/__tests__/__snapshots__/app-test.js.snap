// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
<WithStyles(ForwardRef(Container))
  className="app"
>
  <WithStyles(ForwardRef(Typography))
    variant="h2"
  >
    SWA Todos
  </WithStyles(ForwardRef(Typography))>
  <Connect(Component) />
</WithStyles(ForwardRef(Container))>
`;
